src/blueteam/Robot.java

```diff
@@ -328,4 +343,125 @@ void dodge() {
 		}
 		return Optional.empty();
 	}
+
+	/**
+	 *
+	 * @param map
+	 *            A MapLocation to convert to integer representation
+	 * @return An array arr such that: arr[0] - integer part of x arr[1] -
+	 *         decimal part of x * 10^6 and rounded arr[2] - integer part of y
+	 *         arr[3] - decimal part of y * 10^6 and rounded
+	 */
+	int[] convertMapLocation(MapLocation map) {
```

Comment:
This implementation seems to be overcomplicated to me.  Why could not we use Float.floatToIntBits(val) for serialization and Float.intBitsToFloat(val) for deserialization? It takes half space, it is simpler and more precise...